File: trunk/src/test/java/net/esper/client/TestEPServiceProviderManager.java
package net.esper.client;

import junit.framework.TestCase;

public class TestEPServiceProviderManager extends TestCase
{
    public void testGetInstance()
    {
        Configuration configuration = new Configuration();

        EPServiceProvider runtimeDef1 = EPServiceProviderManager.getDefaultProvider();
        EPServiceProvider runtimeA1 = EPServiceProviderManager.getProvider("A");
        EPServiceProvider runtimeB = EPServiceProviderManager.getProvider("B");
        EPServiceProvider runtimeA2 = EPServiceProviderManager.getProvider("A");
        EPServiceProvider runtimeDef2 = EPServiceProviderManager.getDefaultProvider(configuration);
        EPServiceProvider runtimeA3 = EPServiceProviderManager.getProvider("A", configuration);

        assertNotNull(runtimeDef1);
        assertNotNull(runtimeA1);
        assertNotNull(runtimeB);
        assertTrue(runtimeDef1 == runtimeDef2);
        assertTrue(runtimeA1 == runtimeA2);
        assertTrue(runtimeA1 == runtimeA3);
        assertFalse(runtimeA1 == runtimeDef1);
        assertFalse(runtimeA1 == runtimeB);
    }

    public void testInvalid()
    {
        Configuration configuration = new Configuration();
        configuration.addEventTypeAlias("x", "xxx.noclass");

        try
        {
            EPServiceProviderManager.getProvider("someURI", configuration);
            fail();
        }
        catch (ConfigurationException ex)
        {
            // Expected
        }
    }
}

File: NEsper.Tests/client/TestEPServiceProviderManager.cs
///////////////////////////////////////////////////////////////////////////////////////
// Copyright (C) 2007 Esper Team. All rights reserved.                                /
// http://esper.codehaus.org                                                          /
// ---------------------------------------------------------------------------------- /
// The software in this package is published under the terms of the GPL license       /
// a copy of which has been included with this distribution in the license.txt file.  /
///////////////////////////////////////////////////////////////////////////////////////

using System;

using NUnit.Framework;

namespace net.esper.client
{
	[TestFixture]
	public class TestEPServiceProviderManager
	{
	    [Test]
	    public void testGetInstance()
	    {
	        Configuration configuration = new Configuration();

	        EPServiceProvider runtimeDef1 = EPServiceProviderManager.GetDefaultProvider();
	        EPServiceProvider runtimeA1 = EPServiceProviderManager.GetProvider("A");
	        EPServiceProvider runtimeB = EPServiceProviderManager.GetProvider("B");
	        EPServiceProvider runtimeA2 = EPServiceProviderManager.GetProvider("A");
	        EPServiceProvider runtimeDef2 = EPServiceProviderManager.GetDefaultProvider(configuration);
	        EPServiceProvider runtimeA3 = EPServiceProviderManager.GetProvider("A", configuration);

	        Assert.IsNotNull(runtimeDef1);
	        Assert.IsNotNull(runtimeA1);
	        Assert.IsNotNull(runtimeB);
	        Assert.IsTrue(runtimeDef1 == runtimeDef2);
	        Assert.IsTrue(runtimeA1 == runtimeA2);
	        Assert.IsTrue(runtimeA1 == runtimeA3);
	        Assert.IsFalse(runtimeA1 == runtimeDef1);
	        Assert.IsFalse(runtimeA1 == runtimeB);

	        Assert.AreEqual("A", runtimeA1.URI);
	        Assert.AreEqual("A", runtimeA2.URI);
	        Assert.AreEqual("B", runtimeB.URI);
	        Assert.IsNull(runtimeDef1.URI);
	        Assert.IsNull(runtimeDef2.URI);
	    }

	    [Test]
	    public void testInvalid()
	    {
	        Configuration configuration = new Configuration();
	        configuration.AddEventTypeAlias("x", "xxx.noclass");

	        try
	        {
	            EPServiceProviderManager.GetProvider("someURI", configuration);
	            Assert.Fail();
	        }
	        catch (ConfigurationException ex)
	        {
	            // Expected
	        }
	    }
	}
} // End of namespace
